src/mame/sinclair/sprinter.cpp

```diff
@@ -1506,10 +1518,19 @@ void sprinter_state::do_cpu_wait(bool is_io)
 
 TIMER_CALLBACK_MEMBER(sprinter_state::irq_on)
 {
-	m_maincpu->pulse_input_line(INPUT_LINE_IRQ0, attotime::from_ticks(26, m_maincpu->clock()));
+	if (m_maincpu->input_state(INPUT_LINE_IRQ0) != HOLD_LINE)
+	{
+		m_maincpu->set_input_line(INPUT_LINE_IRQ0, ASSERT_LINE);
+		m_irq_off_timer->adjust(attotime::from_ticks(32, m_maincpu->unscaled_clock()));
+	}
 	update_int(false);
 }
 
+TIMER_CALLBACK_MEMBER(sprinter_state::irq_off)
+{
+	m_maincpu->set_input_line(INPUT_LINE_IRQ0, CLEAR_LINE);
+}
+
```

Comment:
Using `HOLD_LINE` to stop the interrupt pulse from `irq_on` clearing the interrupt from `cbl_tick` seems like a bit of a hack.  Can you keep the state of the two sources and combine them to assert/clear the interrupt line to the CPU?

Reply:
done